extended/src/main/java/apoc/load/LoadXls.java

```diff
@@ -388,14 +405,24 @@ public XLSSpliterator(Sheet sheet, Selection selection, String[] header, String
             this.skip = skip + selection.getOrDefault(selection.top, sheet.getFirstRowNum()) + headerOffset;
             this.limit = limit == Long.MAX_VALUE ? selection.getOrDefault(selection.bottom, sheet.getLastRowNum()) : skip + limit;
             lineNo = this.skip;
+            this.skipNulls = skipNulls;
         }
 
         @Override
         public boolean tryAdvance(Consumer<? super XLSResult> action) {
             try {
                 Row row = sheet.getRow((int)lineNo);
-                if (row != null && lineNo <= limit) {
-                    Object[] list = extract(row, selection);
+                if (lineNo <= limit) {
+
+                    Object[] list;
+                    if (row != null) {
+                        list = extract(row, selection);
+                    } else if (skipNulls && lineNo <= sheet.getLastRowNum()) {
+                        // list with null values (i.e.: empty xls row)
+                        list = new Object[selection.right - selection.left];
```

Comment:
Ma non va bene fare una ricorsione tipo:
```suggestion
                        ++lineNo;
                        return tryAdvance(action)
```